### PR TITLE
Fix CLI to make -team_name optional or not required.  I also added a -join_team cmd.

### PR DESCRIPTION
**Pull request by coreyhulen**

### CODE REVIEW

File: mattermost.go [hunk 1]
```diff
@@ -949,12 +984,10 @@ FLAGS:
 
     -team_name="name"                 The team name used in other commands
 
-    -role="admin"                     The role used in other commands
+    -role="syste_admin"               The role used in other commands
```
Comment (jwilander) [inline mattermost.go]: Typo, should be "system_admin"

**Comment by jwilander** [comment]
+1, other than typo

**Comment by coreyhulen** [comment]
typo fixed

**Comment by crspeller** [comment]
+1

**Pull request merged**